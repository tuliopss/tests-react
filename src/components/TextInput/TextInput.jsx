import React from "react";
import "./TextInput.css";
const TextInput = ({ handleChange, searchValue }) => {
  return (
    <input
      type='search'
      onChange={handleChange}
      value={searchValue}
      className='text-input'
      placeholder='Type your search'
    />
  );
};

export default TextInput;
