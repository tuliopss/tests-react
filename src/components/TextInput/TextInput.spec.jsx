import { describe, expect, it, vi } from "vitest";
import TextInput from "./TextInput";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

// const propsMock = {
//   posts: [
//     {
//       id: 1,
//       title: "title",
//       body: "body 1",
//       cover: "img/img.png",
//     },
//     {
//       id: 2,
//       title: "title 2",
//       body: "body 2",
//       cover: "img/img.png 2",
//     },
//     {
//       id: 3,
//       title: "title 3",
//       body: "body 3",
//       cover: "img/img.png 3",
//     },
//   ],
// };
describe("<TextInput/>", () => {
  it("should have a value of searchValue", () => {
    const fn = vi.fn();
    render(<TextInput handleChange={fn} searchValue={"testadno"} />);

    const input = screen.getByPlaceholderText(/type your search/i);

    expect(input.value).toBe("testadno");
  });

  it("should call handleChange function on each key pressed", async () => {
    const fn = vi.fn();
    const user = userEvent.setup();
    render(<TextInput handleChange={fn} />);

    const input = screen.getByPlaceholderText(/type your search/i);

    const value = "value";

    await user.type(input, value);

    expect(input.value).toBe(value);

    expect(fn).toHaveBeenCalledTimes(value.length);
  });
  it("should match snapshot", () => {
    const fn = vi.fn();
    const { container } = render(<TextInput handleChange={fn} />);

    expect(container).toMatchSnapshot();
  });
});
