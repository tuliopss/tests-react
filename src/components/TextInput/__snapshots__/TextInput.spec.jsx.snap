// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TextInput/> > should match snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="Type your search"
    type="search"
    value=""
  />
</div>
`;
